crypto/sm2/build.info

```diff
@@ -1,5 +1,5 @@
 LIBS=../../libcrypto
 SOURCE[../../libcrypto]=\
-        sm2_sign.c sm2_crypt.c sm2_err.c
+        sm2_sign.c sm2_crypt.c sm2_err.c sm2_key.c
```

Comment:
It's actually strange that none other is needed that way, but anyway, to get `sm2_key.c` included with the FIPS module as well, you must add this:

```
SOURCE[../../providers/libfips.a]=sm2_key.c
```

Reply:
Uhm... But why should there be SM2 code in the fips specific dependencies? It's a bit counterintuitive..

Reply:
Ah!  Good question, and I think you have the answer if you look in `providers/implementations/keymgmt/build.info`, see for example how the RSA keymgmt is included.
(it's not quite done the way I was imagining it, but that's a cleanup that can be done later)

Reply:
Please have a look at 4dfad461e4a28a1b0e82c48f308c7dbba8ad3e3e

I had to decide if I should have a

```C
#if defined(FIPS_MODULE) && !defined(OPENSSL_NO_SM2)
# define OPENSSL_NO_SM2
#endif
```

section at the top of `providers/implementations/keymgmt/ec_kmgmt.c` or (like I ended up doing) adding a define for the libfips.a target.

What I am not sure about is: by setting

```
DEFINE[../../libfips.a]=OPENSSL_NO_SM2
```

am I overwriting the previous definition or adding to it?
If it is overwriting, is it limited to the sources defined in this `build.info` file or affects definitions at a global level?
Does the syntax support `+=` or something to that effect?

Reply:
> It's actually strange that none other is needed that way

I was thinking myself about the reason, and I believe the difference is that the other SM2 specific functions are not referenced from `libimplementations.a` but only from code compiled as part of `libcrypto.a` for the default provider and from code in `libfips.a` which masks references to the SM2 calls using the `FIPS_MODULE` guard.


@slontis on a side note, I noticed the FIPS provider is including the ECX stuff, is that intentional?